### DISCUSSION TITLE
Is it possible to upload the successful screenshot for test evidence to ReportPortal using cypress?

### DISCUSSION THREAD

**Discussion by govindd06** [cypress-io/cypress#25724] [Questions and Help]
In Cypress, it is uploading the screenshot for the failure tests to Reportportal but it is not sending the screenshot for successful tests steps screenshots. Is it possible to configure to capture both success and failure screenshots capturing in the ReportPortal. And also how to upload the videos of only the failure tests to ReportPortal.

**Comment by portyu9** [comment]
Hi @govindd06 

Yes. There are two separate pieces here: **what Cypress captures** and **what the ReportPortal agent uploads**.

For successful tests, Cypress does not automatically take a success screenshot. Take the evidence screenshot explicitly:

```js
cy.screenshot('checkout-success')
```

The current `@reportportal/agent-js-cypress` supports normal Cypress screenshots, so that screenshot can be attached to the ReportPortal result.

For videos, the ReportPortal agent has exactly the options you need:

```js
const { defineConfig } = require('cypress')
const registerReportPortalPlugin =
  require('@reportportal/agent-js-cypress/lib/plugin')

module.exports = defineConfig({
  video: true,

  reporter: '@reportportal/agent-js-cypress',

  reporterOptions: {
    // endpoint, project, apiKey, launch, etc.

    uploadVideo: true,
    uploadVideoForNonFailedSpec: false,
  },

  e2e: {
    setupNodeEvents(on, config) {
      return registerReportPortalPlugin(on, config)
    },
  },
})
```

`uploadVideo: true` enables video uploads, while:

```js
uploadVideoForNonFailedSpec: false
```

means videos are uploaded only for **failed specs**.

That last distinction is important: Cypress records video per **spec run**, not a separate video for each individual test.

Also note that current Cypress defaults `video` to `false`, so you need `video: true` if you want ReportPortal to have Cypress videos available to upload.

Please mark this answer as accepted if it helped, thank you.
